streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -569,7 +570,13 @@ private void runLoop() {
                 log.warn("Detected the states of tasks " + e.corruptedTaskWithChangelogs() + " are corrupted. " +
                              "Will close the task as dirty and re-create and bootstrap from scratch.", e);
 
-                taskManager.commitAll();
+                taskManager.commitInternal(
+                    taskManager.tasks()
+                        .values()
+                        .stream()
+                        .filter(t -> !e.corruptedTaskWithChangelogs().containsKey(t.id()))
+                        .collect(Collectors.toSet())
+                );
```

Comment:
If we hit a `TaskCorruptedException`, we know that only a task in restore mode could be affected and those don't have anything to be committed (their `commitNeeded` flag should be set to `false`). Hence, we just commit all non-corrupted tasks. Afterwards we can safely call `handleCorruption()` (if we don't commit, we might abort a pending transaction for eos-beta incorrectly within `handleCorruption()`)

\cc @abbccdda @guozhangwang